Query by child value

LuisRodriguezLD opened the pull request:
New implementation, works great on iOS. Sadly can't test on Android but should work

LuisRodriguezLD committed:
new query by child value implementation

poul-kg commented:
@Dhvl-Golakiya please review and test this on both iOS and Android and merge if everything works fine.

LuisRodriguezLD committed:
docs fix

LuisRodriguezLD commented:
FYI, I was _finally_ able to test on Android and it works great

Dhvl-Golakiya commented on src/Firebase.Database/Database.uno:
```diff
@@ -656,6 +656,14 @@ namespace Firebase.Database
         }
     }
 
+    internal class ReadByQueryEqualToValue : Promise<string>
```
You have to declare this class as `extern(!mobile)` We are getting an error in Mobile build with a duplicate declaration.

Dhvl-Golakiya commented on src/Firebase.Database/Database.uno:
```diff
@@ -732,4 +772,50 @@ namespace Firebase.Database
+    internal class ReadByQueryEqualToValue : Promise<string>
+    {
+        [Foreign(Language.Java)]
+        public ReadByQueryEqualToValue(string path)
```
Need three parameters `path`, '`key` and `val` for this function.

LuisRodriguezLD committed:
Fixed extern !mobile and arguments in query

I do not know what I was thinking, I have two FuseFirebase repos, must have missed this.
